qiskit_machine_learning/circuit/library/qnn_circuit.py

```diff
@@ -0,0 +1,186 @@
+from qiskit_machine_learning.utils import derive_num_qubits_feature_map_ansatz
+
+
+class QNNCircuit(BlueprintCircuit):
+    """
+    The QNN circuit is a blueprint circuit that wraps feature map and ansatz circuits.
+    It can be used to simplify the composition of these two.
+
+    If only the number of qubits is provided the ``RealAmplitudes`` ansatz ``ZZFeatureMap``
+    are used. If the number of qubits is 1 the ``ZFeatureMap``is used.
+    If only a feature map is provided, the ``RealAmplitudes`` ansatz with the
+    corresponding number of qubits is used. If only an ansatz is provided the ``ZZFeatureMap``
+    with the corresponding number of qubits is used.
+
+    At least one parameter has to be provided. If a feature map and an ansatz is provided, the
+    number of qubits must be the same.
+
+    In case number of qubits is provided along with either a feature map, an ansatz or both, a
+    potential mismatch between the three inputs with respect to the number of qubits is
+    resolved by constructing the QNN circuit with the given number of qubits.
+    If the number of qubits of the QNN circuit needs to be updated after class has been built
+    the number of qubits property has to be used.
+
+    Example:
+
+    .. code-block:: python
+
+        from qiskit_machine_learning.circuit.library import QNNCircuit
+        qnn_qc = QNNCircuit(2)
+        print(qnn_qc)
+        # prints:
+        #      ┌──────────────────────────┐»
+        # q_0: ┤0                         ├»
+        #      │  ZZFeatureMap(x[0],x[1]) │»
+        # q_1: ┤1                         ├»
+        #      └──────────────────────────┘»
+        # «     ┌──────────────────────────────────────────────────────────┐
+        # «q_0: ┤0                                                         ├
+        # «     │  RealAmplitudes(θ[0],θ[1],θ[2],θ[3],θ[4],θ[5],θ[6],θ[7]) │
+        # «q_1: ┤1                                                         ├
+        # «     └──────────────────────────────────────────────────────────┘
+
+        print(qnn_qc.num_qubits)
+        # prints: 2
+
+        print(qnn_qc.input_parameters)
+        # prints: ParameterView([ParameterVectorElement(x[0]), ParameterVectorElement(x[1])])
+
+        print(qnn_qc.weight_parameters)
+        # prints: ParameterView([ParameterVectorElement(θ[0]), ParameterVectorElement(θ[1]),
+        #         ParameterVectorElement(θ[2]), ParameterVectorElement(θ[3]),
+        #         ParameterVectorElement(θ[4]), ParameterVectorElement(θ[5]),
+        #         ParameterVectorElement(θ[6]), ParameterVectorElement(θ[7])])
+    """
+
+    def __init__(
+        self,
+        num_qubits: int | None = None,
+        feature_map: QuantumCircuit | None = None,
+        ansatz: QuantumCircuit | None = None,
+    ) -> None:
+        """
+        Args:
+            num_qubits:  Number of qubits.
+            feature_map: A feature map.
+            ansatz:      An ansatz.
```

Comment:
```
num_qubits:  Optional number of qubits. If not supplied defaults from sizes of feature_map and ansatz.  
    If both feature map and ansatz are left as default then it must be provided.
feature_map: An optional feature map. If not provided defaults to ZZFeatureMap, or ZFeatureMap if
   num qubits is determined to be .
ansatz:      An optional ansatz. If not provided defaults to RealAmplitudes. if not provided.
```
Can we put some description on each of the parameters, say as above. The class docstring has a more wordy description overall but hopefully this summarizes enough.